src/main/java/org/wso2/carbon/module/swiftiso20022/utils/ISOMessageParser.java

```diff
@@ -0,0 +1,116 @@
+
+import java.util.List;
+
+/**
+ * Extract part of the ISO 20022 XML message from message context.
+ */
+public class ISOMessageParser {
+
+    /**
+     * Retrieve specific OMElement from MessageContext according to the XPATH.
+     * @param xPath         XPath expression to the OMElement
+     * @param mc            MessageContext which contains the XML 20022 input
+     * @return OMElement    Required XML element as OMElement
+     * @throws JaxenException
+     */
+    public static OMElement getXMLElementByXPath(String xPath, MessageContext mc) throws JaxenException {
+        AXIOMXPath xpathExp = new AXIOMXPath(xPath);
+
+        // Setting up XML namespaces for AXIOM SOAP, AppHdr, and Document
+        xpathExp.addNamespace(ConnectorConstants.SOAP_PREFIX, ConnectorConstants.SOAP_NAMESPACE);
+        xpathExp.addNamespace(ConnectorConstants.APPHDR_PREFIX, ConnectorConstants.XML_INPUT_APPHDR_NAMESPACE);
+        xpathExp.addNamespace(ConnectorConstants.DOCUMENT_PREFIX, ConnectorConstants.XML_INPUT_DOCUMENT_NAMESPACE);
+
+        OMNode rootElement = mc.getEnvelope().getBody();
+        return (OMElement) xpathExp.selectSingleNode(rootElement);
+    }
+
+    /**
+     * Retrieve specific list of OMElement from MessageContext according to the XPATH.
+     * @param xPath                 XPath expression to the OMElements
+     * @param mc                    MessageContext which contains the XML 20022 input
+     * @return "List<OMELement>"    List of OMElements
+     * @throws JaxenException
+     */
+    public static List<OMElement> getXMLElementsByPath(String xPath, MessageContext mc) throws JaxenException {
+        AXIOMXPath xpathExp = new AXIOMXPath(xPath);
+
+        // Setting up XML namespaces for AXIOM SOAP, AppHdr, and Document
+        xpathExp.addNamespace(ConnectorConstants.SOAP_PREFIX, ConnectorConstants.SOAP_NAMESPACE);
+        xpathExp.addNamespace(ConnectorConstants.APPHDR_PREFIX, ConnectorConstants.XML_INPUT_APPHDR_NAMESPACE);
+        xpathExp.addNamespace(ConnectorConstants.DOCUMENT_PREFIX, ConnectorConstants.XML_INPUT_DOCUMENT_NAMESPACE);
+
+        OMNode rootElement = mc.getEnvelope().getBody();
+        return xpathExp.selectNodes(rootElement);
+    }
+
+    /**
+     * Read the XML input from MessageContext and return the content of child elements of the parent tag as a String.
+     * @param mc         MessageContext which contains the XML 20022 input
+     * @param xPath      Xpath for the XML contents which needs to be extracted
+     * @return           Child contents as a String element
+     */
+    public static String extractISOMessage(MessageContext mc, String xPath) throws ConnectException,
+            OMException, JaxenException  {
+        OMElement element = getXMLElementByXPath(xPath, mc);
+
+        if (element == null) {
+            // No element to be extracted
+            throw new ConnectException("Error: " + xPath + " element not present in the XML");
```

Comment:
lets use string.format or StringUtils to contsruct the error message

Reply:
Updated in a8b4910b6069215e22b8e38302885c5281f9a043